### PR TITLE
Support Force & Confirm

### PR DESCRIPTION
**Pull request by derekbekoe**
Closes https://github.com/Azure/azure-cli/issues/1355

- When registering a command, add `confirmation=True` to enable user confirmation. Also supports a string message or callable.
- Add a —force flag for commands that support this feature.
- Integrated with configuration system so it can be enabled/disabled by setting AZURE_CORE_DISABLE_CONFIRM_PROMPT

Added for the following commands:
az group delete
az vm delete
az network dns zone delete

---

EXAMPLES:

```
$ az group delete -n debekoe-a3
Are you sure you want to perform this operation? (y/n): n
Operation cancelled.
```

Only accepts appropriate values.
```
$ az group delete -n debekoe-a3
Are you sure you want to perform this operation? (y/n): 1
Are you sure you want to perform this operation? (y/n): 2
Are you sure you want to perform this operation? (y/n): 3
Are you sure you want to perform this operation? (y/n): n
Operation cancelled.
```

```
$ az group delete -n debekoe-a3 --force
$
```

No TTY:
```
$ az group delete -n debekoe-a3
Unable to prompt for confirmation as no tty available. Use --force.
Operation cancelled.
```

### CODE REVIEW

**Comment by derekbekoe** [comment]
Only added it for these commands:
az group delete
az vm delete
az network dns zone delete

Are there any other critical ones we should add it for @tjprescott & @johanste?

**Comment by tjprescott** [comment]
I believe the port does this for all delete operations, so I would lean toward that.

File: doc/authoring_command_modules/authoring_commands.md [hunk 1]
```diff
@@ -35,14 +35,15 @@ from azure.cli.commands import cli_command
...
   - `module_name` - The name of the module that is registering the command (e.g. `azure.cli.command_modules.vm.commands`). Typically this will be `__name__`.
   - `name` - String uniquely naming your command and placing it within the command hierachy. It will be the string that you would type at the command line, omitting `az` (ex: access your command at `az mypackage mycommand` using a name of `mypackage mycommand`).
   - `operation` - The handler that will be executed. Format is `<module_to_import>#<attribute_list>`
       - For example if `operation='azure.mgmt.compute.operations.virtual_machines_operations#VirtualMachinesOperations.get'`, the CLI will import `azure.mgmt.compute.operations.virtual_machines_operations`, get the `VirtualMachinesOperations` attribute and then the `get` attribute of `VirtualMachinesOperations`.
   - `table_transformer` (optional) - Supply a callable that takes, transforms and returns a result for table output.
+  - `confirmation` (optional) - Supply True to enable default confirmation. Alternatively, supply a callable that takes the command arguments as a dict and returning a boolean. Alternatively, supply a string for the prompt.
```
Comment (tjprescott) [inline doc/authoring_command_modules/authoring_commands.md]: I'm not entirely sure I understand this. There are three cases:

1) True => default prompt enabled
2) callable => you have access to the command namespace (?) and return true or false? Do we use this?
3) string => enables the prompt with a custom message.

Does using #2 mean you essentially are using your own handler and throwing away the default one?

Comment (derekbekoe) [inline doc/authoring_command_modules/authoring_commands.md]: Your understanding of 1 & 3 is correct.
For 2, you can specify your own confirmation prompt handler. You get the command namespace so you can use it if required. e.g. Print a custom prompt based on the name of one of the parameters.

And no we do not currently use 2 or 3 but there are use-cases where they'll be useful.

**Pull request merged**